reactioncommerce/reaction
# New SortableTable component to replace MeteorGriddle

The original plan was to upgrade our existing Griddle v0.7 to the current v1.5. However, after starting the transition, it doesn't seem the development of Griddle has kept up with our needs. A lot of the basic functionality we used in v0.7 (i.e `onClick` of rows) is missing from v1.5, and after diving deep into the griddle `src`, the replacement Components seem to still be a WIP. I've been in their chat and forums trying to see if there are solutions while still using Griddle, but the community seems to be on the same page as we are. To keep up with React releases, and keep our codebase up to date (and not have BitHound yell at us), we can't wait for Griddle to catch back up.

Therefore, I have instead swapped out Griddle for React Table, which I feel like is a much better, extensible solution for our tables. React Table (9.9kb) is also only ~16% of the size of Griddle (61.7kb), which is always a plus.

I've tried to keep all existing files exactly the same, and only update our wrapper, however I did need to swap out some key names in all of our `customColumnMetadata` variables on each table instance.

- New `<SortableTable />` component as a wrapper around ReactTable
- New `<SortableTableLegacy />` is our wrapper for TacoTable, which is used in Search, but will be / should be swapped out for `<SortableTable />`, if possible. 

**Converted Tables:**
- Email Logs
- Custom Taxes
- Avalara Logs
- Templates list
- Discount Codes
- Discount Rates
- Shipping Custom Rates
- Shipping Shippo

**To Test:**
On the above tables, the only real difference you should notice between the old and new tables is the look / style. Please make sure the filters work, the column sort works, the i18n is working on column headers, the data is populating as it should, and you are able to click on a row and see the editing / more info area display below the table. Please make sure the table / editing is saved if you change a field, and the table reactively updates the data. I also updated the Templates dashboard to use a bigger size of the sidebar, so it was easier to edit the block of HTML.

### NOTE
The functionality is all here, the style of this table is not quite final. I'll still be updating the look of the Pagination and filters (possibly having them done before this is even through the PR process), I just wanted to get a working version merged so we can start to use the functionality throughout the team.

Git: https://github.com/tannerlinsley/react-table
Interesting article: https://medium.com/@tannerlinsley/why-i-wrote-react-table-and-the-problems-it-has-solved-for-nozzle-others-445c4e93d4a8
